widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,461 @@ def data_for_timestamp(self, timestamp):
+    """
+    ROOT_ITEMS = ParametrizedLocator('//div[contains(@class,{@tree_class|quote})]/ul/li/'
+    'span[contains(@class,"expand-icon")]/..')
+    ITEMS_LOCATOR = ParametrizedLocator('//div[contains(@class,{@tree_class|quote})]/ul/li'
+        '/span/../text()[not(contains(.,"Default")) and not(contains(.,"avail"))]/..')
+
+    def __init__(self, parent, tree_class=None, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.tree_class = tree_class
+
+    @property
+    def is_displayed(self):
+        script = "return document.getElementsByClassName('{}').length !== 0".format(self.tree_class)
+        return self.browser.execute_script(script)
+
+    @property
+    def mapping_targets(self):
+        """Returns list of Mapping targets or in other words, leaf nodes."""
+        mapping_targets_list = list()
+        for element in self.browser.elements(self.ROOT_ITEMS.locator):
+                mapping_targets_list.append(element.text)
+        return mapping_targets_list
+
+    @property
+    def mapping_sources(self):
+        """Returns list of Mapping sources or in other words, parent nodes to leaf nodes."""
+        sources_list = list()
+        for element in self.browser.elements(self.ITEMS_LOCATOR.locator):
+                sources_list.append(element.text)
+        return sources_list
+
+    @property
+    def selected_items(self):
+        """Returns list of items that are currently selected."""
+        selected_items_list = list()
+        for element in self.browser.elements(self.ITEMS_LOCATOR.locator):
+            if 'node-selected' in element.get_attribute('class'):
+                selected_items_list.append(element.text)
+        for element in self.browser.elements(self.ROOT_ITEMS.locator):
+            if 'node-selected' in element.get_attribute('class'):
+                selected_items_list.append(element.text)
+        return selected_items_list
+
+    @property
+    def all_items_locators(self):
+        """Returns list of selectors for all items."""
+        tree_elements = dict()
+        for element in self.browser.elements(self.ITEMS_LOCATOR.locator):
+            tree_elements[element.text] = element
+
+        for element in self.browser.elements(self.ROOT_ITEMS.locator):
+            tree_elements[element.text] = element
+        return tree_elements
+
+    def read(self):
+        return self.selected_items
+
+    def fill(self, values):
+        """Click on each value from list of values."""
+        key_error_count = 0
+        if values:
+            for item in values:
+                try:
+                    self.all_items_locators[item].click()
+                except KeyError:
+                    key_error_count += 1
+                    self.logger.debug("Could not find {} in {}"
+                        .format(item, self.all_items_locators.keys()))
+        return (key_error_count == 0 and len(values) > 0)
+
+
+class InfraMappingCommonButtonsSet1(View):
```

Comment:
**REQUIRED**
All these views below should go in `cfme/v2v`

Reply:
I believe @mshriver wanted me to put it in widgetastic_manageiq

Reply:
From my perspective only widely ised widgets and views should go in `widgetastic_manageiq`. These view are specific for v2v and should go in to a respectful module.

Reply:
@kedark3 don't recall recommending that, but regardless, these aren't MIQ widgets, they're specific views for v2v model, so yeah I'd put them in `cfme/v2v`